python/pyspark/sql/session.py

```diff
@@ -557,7 +577,13 @@ def createDataFrame(self, data, schema=None, samplingRatio=None, verifySchema=Tr
         except Exception:
             has_pandas = False
         if has_pandas and isinstance(data, pandas.DataFrame):
-            data, schema = self._convert_from_pandas(data, schema)
+            if self.conf.get("spark.sql.execution.pandas.respectSessionTimeZone").lower() \
```

Comment:
I feel this is a weird config... I think it's acceptable to introduce behavior change during bug fix, like the type inference bug we fixed when converting pandas dataframe to pyspark dataframe.

Reply:
Do you mean we don't need the config and just fix the behavior? cc @gatorsmile